scapy/tools/UTscapy.py

```diff
@@ -301,12 +302,18 @@ def dump_campaign(test_campaign):
                 print("    %s%s" % (c,k)) 
 
 #### COMPUTE CAMPAIGN DIGESTS ####
+if six.PY2:
+    def crc32(x):
+        return "%08X" % (0xffffffff & zlib.crc32(x))
 
-def crc32(x):
-    return "%08X" % (0xffffffff & zlib.crc32(x))
+    def sha1(x):
+         return hashlib.sha1(x).hexdigest().upper()
+else:
+    def crc32(x):
+        return "%08X" % (0xffffffff & zlib.crc32(bytearray(x, "utf8")))
 
-def sha1(x):
-    return sha.sha(x).hexdigest().upper()
+    def sha1(x):
```

Comment:
Is the new indentation needed ?

Reply:
Yes, the change log is a bit messy but the code is 
```
if six.PY2:
    def crc32(x):
        return "%08X" % (0xffffffff & zlib.crc32(x))

    def sha1(x):
         return hashlib.sha1(x).hexdigest().upper()
else:
    def crc32(x):
        return "%08X" % (0xffffffff & zlib.crc32(bytearray(x, "utf8")))

    def sha1(x):
        return hashlib.sha1(x.encode("utf8")).hexdigest().upper()
```